### ISSUE TITLE
Error: device mismatch in conv2d, lhs: Cpu, rhs: Metal { gpu_id: 4294968713 }

### ISSUE THREAD

**Issue by lijingrs** [huggingface/candle#1854]
cargo run --example resnet --release --features metal -- --image test.jpg

<img width="497" alt="image" src="https://github.com/huggingface/candle/assets/20733031/522f6f0b-981e-4f88-9535-103d9875e746">

help to solve this problem! Thanks!

**Comment by lijing5227** [comment]
maybe this similar to https://github.com/huggingface/candle/issues/1838

**Comment by LaurentMazare** [comment]
I just pushed a fix for the device mismatch in #1856 but now it's running into the missing metal kernel for max-pool as in #1838.

**Comment by LaurentMazare** [comment]
Actually the missing maxpool kernel was just added by @tomsanbear in #1863 and now the resnet example properly works on metal.
```bash
$ cargo run --example resnet --release  --features metal -- --image ~/Downloads/tiger.jpeg
loaded image Tensor[dims 3, 224, 224; f32, metal:4294968416]
model built
tiger, Panthera tigris  : 89.93%
tiger cat               : 9.68%
zebra                   : 0.22%
jaguar, panther, Panthera onca, Felis onca: 0.06%
tabby, tabby cat        : 0.03%
```
